src/math/range.cpp

```diff
@@ -3,45 +3,62 @@
 
 #include "math/range.h"
 
-Range::Range(double minimum, double maximum)
+Range::Range(std::optional<double> minimum, std::optional<double> maximum)
 {
-    minimum_ = minimum;
-    maximum_ = maximum;
+    minimum_ = std::min(minimum, maximum);
+    maximum_ = std::max(minimum, maximum);
```

Comment:
Since you're not checking here whether the `std::optional`s contain valid values, its much safer to use your `set()` function, I would say:
```suggestion
    set(minimum, maximum);
```